### ISSUE TITLE
Specified search enegine in setting doesn't work

### ISSUE THREAD

**Issue by Techguyprivate** [brave/browser-android-tabs#2161]
## Description 
I have set duckduckgo as my default serach engine in normal tab.But searching from addressbar, google search enegine is used. 

## Steps to reproduce 

   1. Set duckduckgo as default search engine in setting for normal tabs.
   2. Go to adressabr and search anything.
   3. Google search enegine will be used and results will be seen inside google search.

## Actual result 

Google search is used.
## Expected result

Duckduckgo should be used.
## Issue happens on 
   - Current Play Store version?Yes.
   - Beta build?Yes.

## Device details
   - Install type (ARM, x86): ARM
   - Device (Phone, Tablet, Phablet): Phone, 
   - Android version: 9 pie
   
## Brave version

1.4.2
### Website problems only
- Does the issue resolve itself when disabling Brave Shields? No.
- Is the issue reproducible on the latest version of Chrome? 
NO.
### Additional information

![Record_2019-10-19-15-02-10_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/43615747/67143077-19dd0c80-f285-11e9-8119-9c424fc0d7fa.gif)

**Comment by szaimen** [comment]
I had the same problem. It could resolve it by switching temporarily to another search engine and afterwards back to duckduckgo.

**Comment by Techguyprivate** [comment]
> I had the same problem. I could resolve it by switching temporarily to another search engine and afterwards back to duckduckgo.

No.

Here what I did .

1. clear cache and data , app data. The problem reappear after sometimes and use.

2. Install and uninstall.

3. Change search engine to anything else than google. 

Everytime problems reappear after sometime and using. 

I forgot google for 3 years . Only due to brave now google has my Ip, and three searches. 

How ?

When I , for the first time, search for a thing and noticed it that google is being used instead of duckduckgo

2nd time to confirm the problem. 

3rd time  for to file this bug.

 Why brave and why ? I haven;t used google for a single search or thing for 3 years. Cause google keep things permanently.

**Comment by srirambv** [comment]
Not seeing any issue here. 
![AndroidSearch](https://user-images.githubusercontent.com/17010094/67144772-930a3e80-f248-11e9-932c-6c871fed296d.gif)

Not sure if this is happening because you seem to be logged into google while performing the search. Did you set DDG as search engine during onboarding or did you change it manually after completing onboarding?

**Comment by Techguyprivate** [comment]
No. I didn't visit goolag at all. 

Also,  It shouldn't have done that too even if I was on google . Isn't it ? 

The problem appearing just recently, 4 to 5 days. 

It won't happen immidiately, but suddenly after some days. 

I can confirm that.

I tried to find a reason, but the appearnce of google is random. 

Right now, I can say keep searching, it will appear. 

I woildn't have wasted my time filing an issue and I am well technical . 

Just want to say a non issue thing that There is no place of google and fb in my life.

**Comment by szaimen** [comment]
> Just want to say a non issue thing that There is no place of google and fb in my life.

So why are you logged in to Google if that is really true?

**Comment by srirambv** [comment]
>Also, It shouldn't have done that too even if I was on google . Isn't it ?
>I woildn't have wasted my time filing an issue and I am well technical .

Just trying to understand whats going on. I use an Android device as well and have DDG as default as well but never ran into this kind of issue. So we definitely need to ask questions to figure out whats going on.

> I woildn't have wasted my time filing an issue and I am well technical .

No one is questioning your technical capabilities.

**Comment by Techguyprivate** [comment]
> > Just want to say a non issue thing that There is no place of google and fb in my life.
> 
> So why are you logged in to Google if that is really true?

I have set my search engine to duckduckgo for normal browsing and qwant for private browsing always.

When I searched from addressbar  , google search engine was used, not duckduckgo.

When I have set duckduckgo, google shouldn't be used at all.

Also, I want to clarify that I have not opened google search at all to begin with. It was unexpected and so that is why I think that it is a bug.


Ok. Any doubt and questions ?

**Comment by Techguyprivate** [comment]
> > Also, It shouldn't have done that too even if I was on google . Isn't it ?
> > It won't happen immidiately, but suddenly after some days.
> > I tried to find a reason, but the appearnce of google is random.
> 
> Just trying to understand whats going on. I use an Android device as well and have DDG as default as well but never ran into this kind of issue. So we definitely need to ask questions to figure out whats going on.
> 
> > I woildn't have wasted my time filing an issue and I am well technical .
> 
> No one is questioning your technical capabilities.

Sorry, that was cringe worthy from me. What  I mean by I am little bit technical that I have tested with

1. Clearing browser cache, data, and all app data.

2. Exiting the browser completely, from recent apps and forcibly stopping it from app management in android setting .

3. Uninstalling it and reinstalling it.

Below are the difficulties in reproducing it.

1. It is random . Yes. It uses duckduckgo for sometimes and sometimes google even when duckduckgo is set as default.

2. Going to setting and clicking on duckduckgo for normal browsing, the problem disappears for some time. But after using sometime , the problem of google being used , reappear.

**Comment by srirambv** [comment]
@Techguyprivate  could you confirm if you have any DDG search engine with an `_` at the end similar to what is mentioned in #2160? and selected as default search engine? 

Could you also confirm what locale/region is set on your device?

**Comment by srirambv** [comment]
Also had you set DDG as default search engine before upgrading from previous versions? Was it set as default on both normal and private tab? Did you enable DDG as default for private tab after opening a new private tab and enable it via the prompt?

**Comment by Techguyprivate** [comment]
> @Techguyprivate could you confirm if you have any DDG search engine with an `_` at the end similar to what is mentioned in #2160? and selected as default search engine?
> 
> Could you also confirm what locale/region is set on your device?

No. I have read that issue of  DDG search engine with an _ at the end similar to what is mentioned in #2160 before  filing my issue.


![Screenshot_2019-10-23-11-52-03-65](https://user-images.githubusercontent.com/43615747/67363805-c5d76e00-f58b-11e9-827e-32475a69e98c.png)
![Screenshot_2019-10-23-11-51-53-62](https://user-images.githubusercontent.com/43615747/67363806-c5d76e00-f58b-11e9-993b-24982825b097.png)
![Screenshot_2019-10-23-11-51-35-56_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/43615747/67363807-c5d76e00-f58b-11e9-8e82-9f874704335e.png)
![Screenshot_2019-10-23-11-51-30-80_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/43615747/67363809-c6700480-f58b-11e9-8686-5d8b9c54835b.png)
![Screenshot_2019-10-23-11-48-10-51_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/43615747/67363811-c6700480-f58b-11e9-83ab-412a55f72566.png)
![Screenshot_2019-10-23-11-48-06-06_e4424258c8b8649f6e67d283a50a2cbc](https://user-images.githubusercontent.com/43615747/67363812-c6700480-f58b-11e9-974b-9cf3a765cd4d.png)

**Comment by Techguyprivate** [comment]
> Also had you set DDG as default search engine before upgrading from previous versions? 

Answer : Yes. In any browser , the first thing I do is to change the search engine to either DDG or startpage as a habit.I have not touched the setting of search engine in Brave .It was DDG from start. 


Was it set as default on both normal and private tab? 

Ans: No. DDG as default in normal tab. Qwant in private tab.


Did you enable DDG as default for private tab after opening a new private tab and enable it via the prompt?
 
ans: No. If I remember I changed it only going to setting , not when opening private tab and during the prompt when opening private tab. 

1. I don't think that I have touched the search setting a while ago. It was always DDG . 

2. When I was searching few days ago , it throws google again and again  . Astonished , I looked at the search setting, it was DDG still. 

3. Just tap inside the search engine in normal mode to make sure it was DDG . 

4. Then after sometime, when searching google was used even when DDG was the default search enegine as shown in the video.

5. Setting the search engine to DDG by going to setting, DDG is being used for sometimes. 

6. But after sometime , Google is being used. After closing the browser , searching, randomly , suddenly Google is being used even when DDG is the deafult as shown in the video.

**Comment by Techguyprivate** [comment]
> Not seeing any issue here.
> ![AndroidSearch](https://user-images.githubusercontent.com/17010094/67144772-930a3e80-f248-11e9-932c-6c871fed296d.gif)
> 
> Not sure if this is happening because you seem to be logged into google while performing the search. Did you set DDG as search engine during onboarding or did you change it manually after completing onboarding?

If you want to replicate the bug, don't touch the search engine setting at all. Set it to DDG in normal and Qwant in private and don't touch it at all. Use it for few days . Restart, close,  search , reboot etc.. After sometimes , when searching from addressbar, google will be used even when DDG is set as default in normal tab.

When I faced the problem ,Going to setting, tapping the search engine to DDG, DDG was used . But only after sometime, when searching , randomly, google was being used. 

No need to go to google and to make a search and then search from address bar. I have not went to google.com to search first and then search from the addressbar to replicate the bug. 

Hope it helps.

**Comment by UUSim** [comment]
Same issue is present here. Happened several times now:

- Once a couple of months ago, after upgrading Brave to latest version
- Once yesterday, after rebooting the phone

Both times, I resolved the issue by temporarily selecting a different search engine.
